creator-node/src/services/stateMachineManager/clockUtils.js

```diff
@@ -0,0 +1,130 @@
+    const replicas = Object.keys(replicasToWalletsMap)
+    await Promise.all(
+      replicas.map(async (replica) => {
+        replicasToUserClockStatusMap[replica] = {}
+
+        const walletsOnReplica = replicasToWalletsMap[replica]
+
+        // Make requests in batches, sequentially, to ensure POST request body does not exceed max size
+        for (
+          let i = 0;
+          i < walletsOnReplica.length;
+          i += MAX_BATCH_CLOCK_STATUS_BATCH_SIZE
+        ) {
+          const walletsOnReplicaSlice = walletsOnReplica.slice(
+            i,
+            i + MAX_BATCH_CLOCK_STATUS_BATCH_SIZE
+          )
+
+          const axiosReqParams = {
+            baseURL: replica,
+            url: '/users/batch_clock_status',
+            method: 'post',
+            data: { walletPublicKeys: walletsOnReplicaSlice },
+            timeout: BATCH_CLOCK_STATUS_REQUEST_TIMEOUT
+          }
+
+          // Sign request to other CN to bypass rate limiting
+          const { timestamp, signature } = generateTimestampAndSignature(
+            { spID: SP_ID },
+            DELEGATE_PRIVATE_KEY
+          )
+          axiosReqParams.params = { spID: SP_ID, timestamp, signature }
+
+          let batchClockStatusResp = []
+          let errorMsg
+          try {
+            batchClockStatusResp = (
+              await retry(async () => axios(axiosReqParams), {
+                retries: MAX_USER_BATCH_CLOCK_FETCH_RETRIES
+              })
+            ).data.data.users
+          } catch (e) {
+            errorMsg = e
+          }
+
+          // If failed to get response after all attempts, add replica to `unhealthyPeers` set for reconfig
+          if (errorMsg) {
+            logger.error(
+              `retrieveClockStatusesForUsersAcrossReplicaSet Could not fetch clock values for wallets=${walletsOnReplica} on replica=${replica} ${errorMsg.toString()}`
+            )
+            unhealthyPeers.add(replica)
+          }
+
+          // Add batch response data to aggregate output map
+          batchClockStatusResp.forEach((userClockValueResp) => {
+            const { walletPublicKey, clock } = userClockValueResp
+            replicasToUserClockStatusMap[replica][walletPublicKey] = clock
+          })
+        }
+      })
+    )
+
+    return {
+      replicasToUserClockStatusMap,
+      unhealthyPeers
+    }
+  }
+
+  /**
+   * Make request to given replica to get its clock value for given user
+   * Signs request with spID to bypass rate limits
+   */
+  static async _retrieveClockValueForUserFromReplica(replica, wallet) {
+    const spID = SP_ID
+
+    const { timestamp, signature } = generateTimestampAndSignature(
+      { spID },
+      DELEGATE_PRIVATE_KEY
+    )
+
+    const clockValue = await CreatorNode.getClockValue(
+      replica,
+      wallet,
+      CLOCK_STATUS_REQUEST_TIMEOUT_MS,
+      {
+        spID,
+        timestamp,
+        signature
+      }
+    )
+
+    return clockValue
+  }
+}
```

Comment:
Copy/pasted from [here](https://github.com/AudiusProject/audius-protocol/blob/4c559fd7b2a7e3aa5e4772c5b153260841fe9072/creator-node/src/snapbackSM/snapbackSM.js#L1378-L1402). Only spId is changed to be read directly from config instead of class variable